### PR TITLE
Add matching brackets exercise

### PR DESCRIPTION
**Pull request by kahgoh**
(Also updated config.json to add references to exercise files)

This is one of the [48in24](https://forum.exercism.org/t/48in24-exercise-06-18-matching-brackets/11396) exercises.

### CODE REVIEW

**Review by keiravillekode — COMMENTED** [review]
lgtm

File: config.json [hunk 1]
```diff
@@ -34,223 +38,254 @@
...
         "prerequisites": [],
         "difficulty": 6
+      },
+      {
+        "slug": "matching-brackets",
+        "name": "Matching Brackets",
+        "uuid": "45892c58-05f0-4dc4-ac7d-99babf8d3006",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 1
```
Comment (keiravillekode) [inline config.json:240]: Let's make this difficulty 5 or 6.

**Comment by keiravillekode** [comment]
Let's add an `instructions.append.md`

My suggestion would be to copy [`exercises/practice/rotational-cipher/.docs/instructions.append.md`](https://github.com/exercism/wasm/blob/main/exercises/practice/rotational-cipher/.docs/instructions.append.md)   which gives the student permission to modify the input buffer.

**Comment by kahgoh** [comment]
Thanks for the review @keiravillekode! I have pushed the commits for the difficulty and the adding the `instructions.append.md`. I can't see it here on the PR yet, but I assume they'll become visible when re-opened.

**Review by keiravillekode — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**